src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -228,7 +228,21 @@ module.exports = function(
   }
 
 
-  function markUpdate(workInProgress) {
+  function markUpdateIfNecessary(workInProgress) {
+    const current = workInProgress.alternate;
+    const instance = workInProgress.stateNode;
+    if (current !== null) {
```

Comment:
Everywhere you call this, you already have the instance on the stack and know whether it is an update or a mount. I'd recommend just inlining this stuff. It will be less abstractions to read when you follow the code and easier to optimize because all these field reads and checks disappear.